src/common-libs/auth/permission-decorator.ts

```diff
@@ -0,0 +1,47 @@
+import * as auth from '../auth';
+import { createMethodDecorator, ResolverData } from 'type-graphql';
+import Context from '../../domain-services/Context';
+import {
+  RequiredPermissionsCondition,
+  RequiredPermissionsConditionWithId,
+} from './permissions';
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export function Permission(
+  requiredPermissions: RequiredPermissionsCondition
+): MethodDecorator {
+  return createMethodDecorator(
+    async ({ args, context }: ResolverData<Context>, next) => {
+      let permissions: RequiredPermissionsConditionWithId;
+
+      if (
+        requiredPermissions !== 'anyone' &&
+        requiredPermissions !== 'noone' &&
+        requiredPermissions.type !== 'global'
+      ) {
+        if (!args.id) {
+          throw new Error('Non global permissions require ID field provided');
+        }
+
+        permissions = {
+          id: args.id,
+          ...requiredPermissions,
+        };
+      } else {
+        permissions = requiredPermissions;
+      }
+
+      if (
+        !(await auth.actionIsPermitted(
+          context.models,
+          context.request,
+          permissions
+        ))
+      ) {
+        throw new Error('No permission to perform this action');
```

Comment:
We probably want to start introducing a collection of errors for the different status codes we'll want to throw, so that this can be a 403 for example

Reply:
I wanted to "port" `errors` module from `hpc_service`, but it is heavily tied to `resitfy`, so I didn't do it in this PR. I agree we should do this soon.

Reply:
This is done now